infra/build/functions/sync/main_test.py

```diff
@@ -180,8 +181,9 @@ def test_sync_projects_update(self):
           'test1': '0 8 * * *',
           'test2': '0 7 * * *'
       }, {project.name: project.schedule for project in projects_query})
-      clean = [project.key for project in projects_query]
-      ndb.delete_multi(clean)
+      req = requests.post(
```

Comment:
Do this once in setUp(), rather than doing this in every single test. setUp is guaranteed to always run. If there is an exception in your test, this will never be cleaned up.